// @ngx-progressbar/http-client/__ivy_ngcc__/esm2015/ngx-progressbar-http-client.js
import { Injectable, NgModule } from '@angular/core';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { finalize } from 'rxjs/operators';
import { NgProgress } from '@ngx-progressbar/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@ngx-progressbar/core';
class NgProgressInterceptor {
    /**
     * @param {?} progress
     */
    constructor(progress) {
        this.progress = progress;
        this.inProgressCount = 0;
    }
    /**
     * @param {?} req
     * @param {?} next
     * @return {?}
     */
    intercept(req, next) {
        this.inProgressCount++;
        if (!this.progress.isStarted) {
            this.progress.start();
        }
        return next.handle(req).pipe(finalize(() => {
            this.inProgressCount--;
            if (this.inProgressCount === 0) {
                this.progress.done();
            }
        }));
    }
}
NgProgressInterceptor.ɵfac = function NgProgressInterceptor_Factory(t) { return new (t || NgProgressInterceptor)(ɵngcc0.ɵɵinject(ɵngcc1.NgProgress)); };
NgProgressInterceptor.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: NgProgressInterceptor, factory: NgProgressInterceptor.ɵfac });
/** @nocollapse */
NgProgressInterceptor.ctorParameters = () => [
    { type: NgProgress, },
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgProgressInterceptor, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc1.NgProgress }]; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgProgressHttpClientModule {
}
NgProgressHttpClientModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgProgressHttpClientModule });
NgProgressHttpClientModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgProgressHttpClientModule_Factory(t) { return new (t || NgProgressHttpClientModule)(); }, providers: [
        { provide: HTTP_INTERCEPTORS, useClass: NgProgressInterceptor, multi: true }
    ] });
/** @nocollapse */
NgProgressHttpClientModule.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgProgressHttpClientModule, [{
        type: NgModule,
        args: [{
                providers: [
                    { provide: HTTP_INTERCEPTORS, useClass: NgProgressInterceptor, multi: true }
                ]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

export { NgProgressHttpClientModule, NgProgressInterceptor as ɵa };
